packages/ui-core/src/components/RadioButton/RadioButton.tsx

```diff
@@ -0,0 +1,83 @@
+import * as React from 'react';
+import * as PropTypes from 'prop-types';
+import './RadioButton.less';
+import cnCreate from 'utils/cn';
+
+export interface IRadioButtonProps {
+    /** Radio button value */
+    value: string;
+    /** Radio button name */
+    name?: string;
+    /** Size of radio button label */
+    textSize?: 'small' | 'medium';
+    /** Disabled state of radio button */
+    isDisabled?: boolean;
+    /** Checked state of radio button */
+    isChecked?: boolean;
+    /** Custom class name */
+    className?: string;
+    children?: React.ReactNode;
+    /** OnChange handler */
+    onChange?(value: string): void;
+}
+
+const cn = cnCreate('mfui-radio-button');
+class RadioButton extends React.Component<IRadioButtonProps> {
+    static propTypes = {
+        textSize: PropTypes.oneOf(['small', 'medium']),
+        isDisabled: PropTypes.bool,
+        isChecked: PropTypes.bool,
+        onChange: PropTypes.func,
+        name: PropTypes.string,
+        value: PropTypes.string.isRequired,
+        children: PropTypes.node,
+        className: PropTypes.string,
+    };
+
+    static defaultProps: Partial<IRadioButtonProps> = {
+        textSize: 'medium',
+        isDisabled: false,
+    };
+
+    handleChange = () => {
+        const { onChange, value } = this.props;
+
+        this.setState({ selectedButton: value });
+        onChange && onChange(value);
+    }
+
+    renderTextElem() {
+        const { textSize } = this.props;
+
+        return <div className={cn('text', { 'size': textSize})}>{this.props.children}</div>;
+    }
+
+    render() {
+        const { isChecked, isDisabled, name, value, children } = this.props;
+        const inputDynamicProps = isChecked !== undefined ? {checked: isChecked} : {};
```

Comment:
```suggestion
        const checkedProp = isChecked !== undefined ? { checked: isChecked } : undefined;
```

На счет undefined в конце не уверен, возможно только твой вариант с {} сработает.

Reply:
почему undefined не сработает. Ведь предусмотрены кейсы, когда пропс isChecked не передается в компонент, а это означает, что вместо него приходит undefined

Reply:
Проверил, undefined сработает

В моем комменте еще правки есть